feat : User 생성 기능

jinseong205 opened the pull request:
feat: 유저 생성 기능
- 유저 생성 dto CreateUserRequest 작성
- controller, service 빈 등록 및 기능 구현
- repository  jpaRepository 상속 및 기능 구현
- 유저 생성 단위 테스트

> @CreatedDate, @column(updatable = false)
> @LastModifiedDate 

#15 내용 적용 이후 생성 일자 / 수정 일자는 자동으로 추적 관리 할 수 있게 entity에  적용

resolved #7

dudwns9280 committed:
feat : User 도메인 폴더 구조 생성 및 User entity 설계
User 도메인 폴더 생성 후 controller, dto, service, repository, entity로 나눠 임시 파일 생성.
@jinseong205 님과 논의한 내용을 토대로 User Entity 설계 완료.
resolves: #5

jinseong205 committed:
feat : User 생성 기능 구현

유저 생성 dto 추가. validation 추가
controller , service각 anotation을 이용하여 빈으로 등록 및 유저 생성 기능 구현
jpaRepostiroy 상속 및 유저 생성 기능 구현
resolves: #7

jinseong205 committed:
feat: 유저 생성 기능

controller, service 빈 등록 및 기능 구현
repository jpaRepository 상속 및 기능 구현

tableMinPark commented:
#18 머지전에 H2에서 User 테이블 생성 관련해서 발생한 이슈 먼저 확인해야할 것 같습니다.

jinseong205 committed:
Merge branch 'develop' of https://github.com/daengdaengLee/until into feat/CreateUser

jinseong205 committed:
feat: user 생성 기능 개발

feat: 유저 생성 기능

유저 생성 dto CreateUserRequest 작성
controller, service 빈 등록 및 기능 구현
repository jpaRepository 상속 및 기능 구현
usreservice createUser 단위테스트 추가

#18 user table 이슈 해결

resolved #7

daengdaengLee commented on src/main/java/hello/until/user/dto/request/CreateUserRequest.java:
```diff
@@ -0,0 +1,15 @@
+import jakarta.validation.constraints.Size;
+import lombok.Data;
+
+@Data
```
이건 getter, setter 만 붙이는 게 어떨까요?

jinseong205 replied:
해당 부분 수정해놓도록 하겠습니다
어제 회의때 나온내용이였는데 수정을 안해놨었네요

daengdaengLee commented on src/main/java/hello/until/user/controller/UserController.java:
```diff
@@ -1,4 +1,26 @@
+	private final UserService userService;
+	
+	@PostMapping("/join")
+	public ResponseEntity<?> join(@RequestBody @Valid CreateUserRequest createUserRequest){
```
응답 JSON 도 간단하게 보내주는 게 어떨까요?

jinseong205 replied:
단순 생성, 수정, 삭제에 발생할 경우 간단한 메세지를 보내는 것이 좋다고 생각합니다.
형식을 정해지면 수정해놓도록 하겠습니다.

daengdaengLee commented on src/main/java/hello/until/user/repository/UserRepository.java:
```diff
@@ -1,4 +1,11 @@
+
+public interface UserRepository extends JpaRepository<User, Long> {
+
+	User findByEmail(String email);
```
응답을 `Optional<User>` 로 받는 건 어떨까요?

daengdaengLee replied:
```suggestion
Optional<User> findByEmail(String email);
```

daengdaengLee commented on src/main/java/hello/until/user/service/UserService.java:
```diff
@@ -1,4 +1,41 @@
+		
+	}
+	
+	@Transactional(readOnly = true)
+	public void validateUser(String eamil) {
```
1. 가시성을 private 으로 해도 되지 않을까요?
2. this.xxx 로 내부 호출할 때는 @Transactional 이 안 먹는 걸로 알고 있어요. 외부에서 호출할 일이 없으면 애노테이션 제거해도 되지 않을까요?

jinseong205 replied:
해당 내용에 대해 생각 못하고 있었습니다. 수정 할게요 감사합니다!

daengdaengLee commented on src/test/java/hello/until/user/service/UserServiceTest.java:
```diff
@@ -0,0 +1,49 @@
+
+		userService.createUser(email, password);
+
+		Mockito.verify(this.userRepository, Mockito.times(1)).save(Mockito.any(User.class));
```
email, password 가 Repository 로 제대로 넘어갔는지도 검사하면 좋을 것 같아요!

jinseong205 replied:
해당 부분 조금 더 찾아서 수정 하도록 하겠습니다

daengdaengLee reviewed (dismissed):
LGTM!

dudwns9280 commented on src/main/java/hello/until/user/controller/UserController.java:
```diff
@@ -1,4 +1,26 @@
+import lombok.RequiredArgsConstructor;
+
+@RestController
+@RequiredArgsConstructor
```
URL 주소를 RequestMapping을 통해서 Controller별로 명시하면 좋을 꺼 같아요

jinseong205 committed:
feat : review 내용 수정

- requse : data -> getter,setter
- service : 접근제한자 수정 , 불필요 transacitonal 제거
- test : respository args 값 확인

#7

daengdaengLee commented on src/main/java/hello/until/user/service/UserService.java:
```diff
@@ -1,4 +1,42 @@
+		
+	}
+	
+	private Optional<User> validateUser(String eamil) {
+		User user = userRepository.findByEmail(eamil);
+		return Optional.ofNullable(user);
+		
+	}
```
여기는 리턴 타입을 boolean 으로 하는 게 좋을 것 같아요~

jinseong205 committed:
feat: 유저 컨트롤러 request mapping 추가

유저 컨트롤러 request mapping 추가

daengdaengLee commented on src/main/java/hello/until/user/controller/UserController.java:
```diff
@@ -1,4 +1,28 @@
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/user")
```
URL 에는 복수형으로 users 가 어떨까요?

jinseong205 replied:
좋은것 같습니다. 해당 내용으로 변경할게요!

jinseong205 committed:
feat: createUser 수정

repository - findByEmail을 Optional<User>로 반환

daengdaengLee reviewed (approved):
LGTM!
